packages/flutter/lib/src/widgets/form.dart

```diff
@@ -765,7 +765,7 @@ class FormFieldState<T> extends State<FormField<T>> with RestorationMixin {
 }
 
 /// Used to configure the auto validation of [FormField] and [Form] widgets.
-enum AutovalidateMode {
+enum AutoValidateMode {
```

Comment:
The ideal solution would probably be

```dart
@Deprecated(...)
typedef AutovalidateMode = AutoValidateMode;

enum AutoValidateMode { ... }
```

so that people don't get a compile-time errors after doing `flutter update`.

It'd also make it so this PR has much fewer diffs!

Reply:
@itsjatinnagar Thank you for bringing that up. This is such an elegant solution. I have reverted my commit and made appropriate changes.